flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/AggregationCodeGenerator.scala

```diff
@@ -356,6 +486,9 @@ class AggregationCodeGenerator(
            """.stripMargin
 
       if (needMerge) {
+        if (accConfig != null && accConfig.isStateBackedDataViews) {
+          throw new CodeGenException("DataView doesn't support merge when the backend uses state.")
```

Comment:
Can we throw this exception earlier (e.g., in `AggregateUtil`) and give more details about the aggregation function?

Reply:
I found the exception will be throw in everywhere call generateAggregations(..).
I think we can print funcName in exception info.